### PR TITLE
Add initial italian rules

### PR DESCRIPTION
**Pull request by alex179ohm**
Fix english rules with new rules name (replace alphanumeric with letter) and adds an initial italian rules file.

### CODE REVIEW

**Comment by Mte90** [comment]
In our discussion we said to try with a first test and see if we need more symbols to exclude as example.

**Comment by alex179ohm** [comment]
@MichaelKohler yes the italian file rules is the same as the english one, maybe after some tests, we start adjusting some rules, the english file rules seems a good start to implement the italian scraper.
incoming commits might contain more accurate italian rules.

> With #18 this will get easier as we will be able to run this in a step in the cloud, until then testing is manual though.
very appreciated :)

>

**Comment by Mte90** [comment]
I got that error:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { inner: ErrorInner { kind: Custom, line: None, col: 0, message: "missing field `min_alphanumeric_characters`", key: [] } }', src/libcore/result.rs:1051:5
```

**Comment by Mte90** [comment]
I added to silent errors:
```
min_alphanumeric_characters = 1
quote_start_with_alphanumeric = true
needs_alphanumeric_start = true
```
I think that the issue happens also in the other languages because they don't have that rule.

**Comment by MichaelKohler** [comment]
Please merge latest master in your branch, the rename is there: https://github.com/Common-Voice/common-voice-wiki-scraper/pull/27/files

**Comment by Mte90** [comment]
I am testing but I see that export a lot of names that I don't think are useful a lot:
```
Virgil Jacomini
Donald Johnston
Clyde King
Alden Sanborn
Alfred Lindley
James Stillman Rockefeller
Frederick Sheffield
John Brinck
Marvin Stalder
Autore del brano è Michel Berger.
David Dunlap
Duncan Gregg
Burton Jastram
Joe Rantz
Frank Shakespeare
William Fields
Molti dei suoi libri sono ispirati ai suoi viaggi in Medio Oriente.
Africa e Asia.
James Dunbar
Wayne Frye
Charles Manring
Richard Murphy
```

There are also other words not very useful:
```
Es.
F. H.
a.C.
Tom M.
ingl.
All'art.
È solitamente abbreviato con il simbolo ©.
Volsinia di E.
In altre parole: se la v.c.
Chaco
Città dell'.
Slavs!
Iser
```

Maybe we can do a regex that ignore this cases?

**Comment by alex179ohm** [comment]
The latest commit should fix some unwanted behaviors (match-name cases).
And fix some clippy warnings on checker.rs and extractor.rs.

File: src/rules/italian.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+min_trimmed_length = 3
+min_word_count = 1
```
Comment (Mte90) [inline src/rules/italian.toml]: I think that is better to put 2, so we can avoid cases:
```
Es:
Algol
```

File: src/rules/italian.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+max_word_count = 14
+min_characters = 0
+may_end_with_colon = false
+quote_start_with_letter = true
+needs_punctuation_end = false
+needs_letter_start = true
+needs_uppercase_start = false
```
Comment (Mte90) [inline src/rules/italian.toml]: True so we can avoid broken sentences

**Comment by Mte90** [comment]
I am wondering if it is possible to put a text characters limit, because for sentence collector and cv, we had a limit of 125b chars per sentence.

**Comment by Mte90** [comment]
We need something to ignore sentences like:
```
Siano "A" = {"a","a","a"} e "B" = {"b","b"}.
Se l'esponente "n" è pari, la disuguaglianza è valida per "ogni" numero reale "x".
La disuguaglianza può anche essere generalizzata ad un qualsiasi esponente reale "r".
In altre parole: se la v.c.
Il nome "disprosio" deriva dal greco "dysprositos", "difficile arrivarci".
Il primo romanzo, "Dirk Gently.
```

I think that for the `"` we can do a regex that ignore sentences with ", so we can avoid maths stuff but also wrong sentences like the last one (usually happen with film titles that are truncated) and for the `v.c.` case if there are more then one point to ignore the sentence.

Anyway now is working better then before, there is only the issue of english surname that sometimes are very difficult.

**Comment by Mte90** [comment]
There are also letters not from the italian accents because are words from other languages that we don't use and can create issues:
```
Una gigantessa poteva anche essere detta "gýgr".
Attacchi aerei anche su Grödig, Hallein, Bischofshofen, Schwarzach.
Deșteaptă-te, Române!
Il testo di "Deșteaptă-te, Române!
Il suo simbolo non ufficiale è Λ.
```
Maybe we need to see the alphabet with the italian accented letter included and we are done.

And also sentences with " without ending, probably is an issue also for other languages:
```
L'anno seguente fu acquistata dalla "Commodore International Ltd.
Esso non è un bene per nessuno"".
Nemini nimium bene est significa letteralmente: ""perché aspiriamo al superfluo?
```

**Comment by Mte90** [comment]
OPened a ticket about the " issues https://github.com/Common-Voice/common-voice-wiki-scraper/issues/32
So we can focus on the alphabet to use, we don't need the full latin alphabet.

**Comment by Mte90** [comment]
Now is perfect we need only to define the alphabet and symbols.
We have still issues like: `Nella sua recensione nel "The New Yorker", la Kael affermò "È un grande film.` but I think that is something that need to be fixed in the scraper itself.

If we change the references to letters with: `[a-zA-ZàèéìíîòóùúÀÈÉÌÍÎÒÓÙÚ]` I think that we are done, for other languages there is https://www.rexegg.com/regex-interesting-character-classes.html

**Comment by nukeador** [comment]
Can we make this PR which with modifications to the Italian file so I can merge?

Thanks!

**Comment by Mte90** [comment]
There is the settings to disable symbols https://github.com/Common-Voice/common-voice-wiki-scraper/blob/master/src/rules/german.toml I think that we can start from there are and add the german (umlaut) and other maths symbols

**Comment by Mte90** [comment]
```
disallowed_symbols = [
  '<', '>', '+', '*', '\', '#', '@', '^', '[', ']', '(', ')', '/', '»', '«',
  'â', 'ă', 'ș', 'ç', 'Å', 'ý', 'ö', 'ā', 'ü', 'ö', '%', '&',
  'α', 'β', 'Γ', 'γ', 'Δ', 'δ', 'ε', 'ζ', 'η', 'Θ', 'θ', 'ι', 'κ',
  'Λ', 'λ', 'μ', 'ν', 'Ξ', 'ξ', 'Π', 'π', 'ρ', 'Σ', 'σ', 'ς', 'τ',
  'υ', 'Φ', 'φ', 'χ', 'Ψ', 'ψ', 'Ω', 'ω', '©'
]
```
And we are done

**Comment by Mte90** [comment]
ready to be merged :-D

**Comment by nukeador** [comment]
This PR also modifies other files, can you make sure it only modifies the Italian rules?

Thanks!

**Comment by Mte90** [comment]
@alex179ohm is a Rust developer (fron the Rome meetup) and I guess that he did that to improve the code quality and avoid issues in the future, do you prefer a second pr for this (tiny) changes?

**Comment by nukeador** [comment]
Yes, please.

**Comment by Mte90** [comment]
done

**Pull request merged**